stubs/astropy/astropy/units/core.pyi

```diff
@@ -0,0 +1,205 @@
+from _typeshed import Incomplete
+from typing import Any
+from typing_extensions import TypeAlias
+
+from astropy.units.equivalencies import Equivalency
+from astropy.utils.exceptions import AstropyWarning
+
+_Equivalences: TypeAlias = list[Equivalency]
+_Aliases: TypeAlias = dict[str, UnitBase]
+
+class _UnitRegistry:  # undocumented
+    def __init__(self, init=..., equivalencies: _Equivalences = ..., aliases: _Aliases = ...) -> None: ...
+    @property
+    def registry(self): ...
+    @property
+    def all_units(self) -> set[UnitBase]: ...
+    @property
+    def non_prefix_units(self) -> set[UnitBase]: ...
+    def set_enabled_units(self, units) -> None: ...
+    def add_enabled_units(self, units) -> None: ...
+    def get_units_with_physical_type(self, unit): ...
+    @property
+    def equivalencies(self) -> _Equivalences: ...
+    def set_enabled_equivalencies(self, equivalencies: _Equivalences) -> None: ...
+    def add_enabled_equivalencies(self, equivalencies: _Equivalences) -> None: ...
+    @property
+    def aliases(self) -> _Aliases: ...
+    def set_enabled_aliases(self, aliases: _Aliases) -> None: ...
+    def add_enabled_aliases(self, aliases: _Aliases) -> None: ...
+
+class _UnitContext:  # undocumented
+    def __init__(self, init=..., equivalencies: _Equivalences = ...) -> None: ...
+    def __enter__(self) -> None: ...
+    def __exit__(self, type: object, value: object, tb: object) -> None: ...
+
+def get_current_unit_registry() -> _UnitRegistry: ...
+def set_enabled_units(units) -> _UnitContext: ...
+def add_enabled_units(units) -> _UnitContext: ...
+def set_enabled_equivalencies(equivalencies: _Equivalences) -> _UnitContext: ...
+def add_enabled_equivalencies(equivalencies: _Equivalences) -> _UnitContext: ...
+def set_enabled_aliases(aliases: _Aliases) -> _UnitContext: ...
+def add_enabled_aliases(aliases: _Aliases) -> _UnitContext: ...
+
+class UnitsError(Exception): ...
+class UnitScaleError(UnitsError, ValueError): ...
+class UnitConversionError(UnitsError, ValueError): ...
+class UnitTypeError(UnitsError, TypeError): ...
+class UnitsWarning(AstropyWarning): ...
+
+class UnitBase:
+    __array_priority__: int
+    def __deepcopy__(self, memo: Any) -> UnitBase: ...
+    def __bytes__(self) -> bytes: ...
+    @property
+    def names(self) -> None: ...
+    @property
+    def name(self) -> None: ...
+    @property
+    def aliases(self) -> None | list[str]: ...
+    @property
+    def scale(self) -> float: ...
+    @property
+    def bases(self) -> list[UnitBase]: ...
+    @property
+    def powers(self) -> list[float]: ...
+    def to_string(self, format=...): ...
+    def __format__(self, format_spec) -> str: ...
+    def __pow__(self, p: float) -> CompositeUnit: ...
+    def __truediv__(self, m: Any): ...
+    def __rtruediv__(self, m: Any): ...
+    def __mul__(self, m: Any): ...
+    def __rmul__(self, m: Any): ...
+    def __rlshift__(self, m: Any): ...
+    def __rrshift__(self, m: Any): ...
```

Comment:
This always returns NotImplemented; again, omitting it from the stub may be more useful.